postinstall fail on certificate issue

trying to install this package for the first time and it fails due to postinstall script and certificate issue.  i can bypass the postinstall script by using MONGOMS_DISABLE_POSTINSTALL but every developer will face the same issue and i'd like to avoid it.
possible suggestions:

1. make postinstall optional
2. have some options set in package.json to overcome this issue (i saw there is a thread about it)


and to the issue at hand....
what i need to do in order to solve the "Error: self signed certificate in certificate chain"  error?
i'd appreciate if this is not a manual process that needs to be set on the env as i'm not the only one working on the source code.
i'm using the package for testing, so it should be simple: clone src, run test....

Info: 
local dev machine: win10
CI/CD: Linux based
node: 10.15
mongodb-memory-server: "^4.2.1"

**********************************************************

```
>npm install mongodb-memory-server --save-dev

> mongodb-memory-server@4.2.1 postinstall [src]\node_modules\mongodb-memory-server
> node ./scripts/postinstall.js

mongodb-memory-server: checking MongoDB binaries cache...
events.js:167
      throw er; // Unhandled 'error' event
      ^

Error: self signed certificate in certificate chain
    at TLSSocket.onConnectSecure (_tls_wrap.js:1049:34)
    at TLSSocket.emit (events.js:182:13)
    at TLSSocket._finishInit (_tls_wrap.js:631:8)
Emitted 'error' event at:
    at TLSSocket.socketErrorListener (_http_client.js:392:9)
    at TLSSocket.emit (events.js:182:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mongodb-memory-server@4.2.1 postinstall: `node ./scripts/postinstall.js`
npm ERR! Exit status 1
```


*********************
Thanks

--- a/packages/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts
+++ b/packages/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts
@@ -31,6 +31,7 @@ interface HttpDownloadOptions {
   port: string;
   path: string;
   method: 'GET' | 'POST';
+  rejectUnauthorized?: boolean;
   agent: HttpsProxyAgent | undefined;
 }
 
@@ -145,6 +146,8 @@ export default class MongoBinaryDownload {
       process.env.HTTPS_PROXY ||
       process.env.HTTP_PROXY;
 
+    const strictSsl = process.env['npm_config_strict-ssl'] === 'false' ? false : true;
+
     const urlObject = url.parse(downloadUrl);
 
     if (!urlObject.hostname || !urlObject.path) {
@@ -156,6 +159,7 @@ export default class MongoBinaryDownload {
       port: urlObject.port || '443',
       path: urlObject.path,
       method: 'GET',
+      rejectUnauthorized: strictSsl,
       agent: proxy ? new HttpsProxyAgent(proxy) : undefined,
     };
 

--- a/packages/mongodb-memory-server-core/src/util/__tests__/MongoBinaryDownload-test.ts
+++ b/packages/mongodb-memory-server-core/src/util/__tests__/MongoBinaryDownload-test.ts
@@ -53,6 +53,32 @@ MONGOMS_MD5_CHECK environment variable`, () => {
     expect(callArg1.agent.proxy.href).toBe('http://user:pass@proxy:8080/');
   });
 
+  it('should not reject unauthorized when strict-ssl is false in env vars', async () => {
+    process.env['npm_config_strict-ssl'] = 'false';
+
+    const du = new MongoBinaryDownload({});
+    du.httpDownload = jest.fn();
+
+    await du.download('https://fastdl.mongodb.org/osx/mongodb-osx-ssl-x86_64-3.6.3.tgz');
+    expect(du.httpDownload).toHaveBeenCalledTimes(1);
+    const callArg1 = (du.httpDownload as jest.Mock).mock.calls[0][0];
+    expect(callArg1.rejectUnauthorized).toBeDefined();
+    expect(callArg1.rejectUnauthorized).toBe(false);
+  });
+
+  it('should reject unauthorized when strict-ssl is not in env vars', async () => {
+    delete process.env['npm_config_strict-ssl'];
+
+    const du = new MongoBinaryDownload({});
+    du.httpDownload = jest.fn();
+
+    await du.download('https://fastdl.mongodb.org/osx/mongodb-osx-ssl-x86_64-3.6.3.tgz');
+    expect(du.httpDownload).toHaveBeenCalledTimes(1);
+    const callArg1 = (du.httpDownload as jest.Mock).mock.calls[0][0];
+    expect(callArg1.rejectUnauthorized).toBeDefined();
+    expect(callArg1.rejectUnauthorized).toBe(true);
+  });
+
   it(`makeMD5check returns true if md5 of downloaded mongoDBArchive is
 the same as in the reference result`, () => {
     const someMd5 = 'md5';

